core/src/main/java/org/jboss/gwt/elemento/core/BodyObserver.java

```diff
@@ -0,0 +1,149 @@
+        mutationObserverInit.setChildList(true);
+        mutationObserverInit.setSubtree(true);
+
+        mutationObserver.observe(document.body, mutationObserverInit);
+    }
+
+    private BodyObserver() {
+
+    }
+
+    private static void onElementsAppended(MutationRecord record) {
+        List<ElementObserver> observed = new ArrayList<>();
+        for (int i = 0; i < attachObservers.size(); i++) {
+            ElementObserver elementObserver = attachObservers.get(i);
+            if (isNull(elementObserver.observedElement())) {
+                observed.add(elementObserver);
+            } else {
+                if (isAppended(elementObserver.attachId())) {
+                    elementObserver.callback().onObserved(record);
+                    elementObserver.observedElement().removeAttribute(ATTACH_UID_KEY);
+                    observed.add(elementObserver);
+                }
+            }
+        }
+
+        attachObservers.removeAll(observed);
+    }
+
+    private static boolean isAppended(String attachId) {
+        return nonNull(DomGlobal.document.body.querySelector("[" + ATTACH_UID_KEY + "='" + attachId + "']"));
+    }
+
+    private static void onElementsRemoved(MutationRecord record) {
+        List<ElementObserver> observed = new ArrayList<>();
+        for (int i = 0; i < detachObservers.size(); i++) {
+            ElementObserver elementObserver = detachObservers.get(i);
+            if (isNull(elementObserver.observedElement())) {
+                observed.add(elementObserver);
+            } else {
+                if (record.removedNodes.asList().contains(elementObserver.observedElement()) || isChildOfRemovedNode(record, elementObserver.attachId())) {
+                    elementObserver.callback().onObserved(record);
+                    elementObserver.observedElement().removeAttribute(DETACH_UID_KEY);
+                    observed.add(elementObserver);
+                }
+            }
+        }
+
+        detachObservers.removeAll(observed);
+    }
+
+    private static boolean isChildOfRemovedNode(MutationRecord record, final String detachId) {
+        List<Node> nodes = record.removedNodes.asList();
+
+        for (int i = 0; i < nodes.size(); i++) {
+            Node node = Js.uncheckedCast(nodes.get(i));
+            if(Node.ELEMENT_NODE== node.nodeType) {
+                if (nonNull(node.querySelector("[" + DETACH_UID_KEY + "='" + detachId + "']"))) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Helper method to register a callback when the element is removed from the document body
+     * note that the callback will be called only once, if the element is removed and re-appended a new callback should be registered.
+     *
+     * @param element
+     * @param callback {@link ObserverCallback}
+     */
+    public static void onDetach(HTMLElement element, ObserverCallback callback) {
```

Comment:
IMO the public methods `onDetach()` and `onAttach()` should be either in `Elements` *or* `BodyObserver`. Although doing the exact same thing, I don't like ambiguity. I prefer to have them in `Elements` and have only package private code in `BodyObserver`. WDYT?

Reply:
I think you are right .. i will work on this.